cpp/include/cuspatial/detail/utility/device_atomics.cuh

```diff
@@ -14,6 +14,61 @@
  * limitations under the License.
  */
 
+#pragma once
+
+#include <algorithm>
+#include <type_traits>
+
+namespace {
+
+template <typename T>
```

Comment:
I would add comments that we need these functions defined here to resolve the otherwise ambiguous lookup of `std::[min|max]` in the template functions below.

Reply:
done